connectors/src/connectors/slack/index.ts

```diff
@@ -353,20 +361,40 @@ export async function retrieveSlackConnectorPermissions({
       break;
   }
 
-  const slackChannels = await SlackChannel.findAll({
-    where: filterPermission
-      ? {
-          connectorId: connectorId,
-          permission: { [Op.or]: permissionToFilter },
-        }
-      : {
-          connectorId: connectorId,
-        },
-    order: [
-      ["createdAt", "DESC"],
-      ["slackChannelName", "ASC"],
-    ],
-  });
+  const slackChannels: {
+    slackChannelId: string;
+    slackChannelName: string;
+    permission: ConnectorPermission;
+  }[] = [];
+
+  const accessToken = await getAccessToken(c.connectionId);
+  const remoteChannels = await getChannels(accessToken, false);
+  for (const remoteChannel of remoteChannels) {
```

Comment:
Let's avoid doing N queries in DB sequentially (N can be large here).

If permission_filter is empty => get all slack channels from Slack
Otherwise => query in DB

Reply:
>If permission_filter is empty => get all slack channels from Slack
> Otherwise => query in DB

I think it's a better design to show remote data as this ensures we have up to date data. If we had to handle all the webhooks  to sync the list of channels, we would add quite some complexity for a screen that is rarely shown to the end user, I don't think it's worth it.

Reply:
Sure but do in parallel the slack call and the db call  and then handle the data. Don't do N SQL calls?

Reply:
Sounds good to me.